dot/types/grandpa.go

```diff
@@ -193,3 +193,45 @@ type GrandpaVote struct {
 func (v *GrandpaVote) String() string {
 	return fmt.Sprintf("hash=%s number=%d", v.Hash, v.Number)
 }
+
+// GrandpaEquivocation is a custom vdt type for a grandpa equivocation
+type GrandpaEquivocation scale.VaryingDataType
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (ge *GrandpaEquivocation) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*ge)
+	err = vdt.Set(val)
+	if err != nil {
+		return err
+	}
+	*ge = GrandpaEquivocation(vdt)
+	return nil
+}
+
+// Value will return the value from the underlying VaryingDataType
+func (ge *GrandpaEquivocation) Value() (val scale.VaryingDataTypeValue, err error) {
+	vdt := scale.VaryingDataType(*ge)
+	return vdt.Value()
+}
+
+// NewGrandpaEquivocation returns a new VaryingDataType to represent a grandpa Equivocation
+func NewGrandpaEquivocation() *GrandpaEquivocation {
+	vdt, err := scale.NewVaryingDataType(PreVoteEquivocation{}, PreCommitEquivocation{})
+	if err != nil {
+		panic(err)
```

Comment:
opinion: return is better than panic

last line in go proverbs https://go-proverbs.github.io/

Reply:
In other use cases on VDT constructors in the code I see that we use Must... so I changed it to that